Sources/MapboxMaps/Foundation/Events/Payload/ResourceRequestPayload.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+
+public struct ResourceRequestPayload: Decodable {
+    public let dataSource: ResourceDataSource
+    public let request: ResourceRequest
+    public let response: ResourceResponse?
+}
+
+public enum ResourceDataSource: String, Decodable {
+    case resourceLoader = "resource-loader"
+    case network
+    case database
+    case asset
+    case fileSystem = "file-system"
+}
+
+public struct ResourceRequest: Decodable {
+    public enum Kind: String, Decodable {
+        case unknown
+        case style
+        case source
+        case tile
+        case glyphs
+        case spriteImage = "sprite-image"
+        case spriteJSON = "sprite-json"
+        case image
+    }
+    public enum Priority: String, Decodable {
+        case regular
+        case low
+    }
+    public enum LoadingMethod: String, Decodable {
+        case cache
+        case network
+    }
+
+    let url: String
+    let kind: Kind
+    let priority: Priority
+    let loadingMethod: [LoadingMethod]
+}
+
+public struct ResourceResponse: Decodable {
+    public enum Source: String, Decodable {
+        case network
+        case cache
+        case tileStore = "tile-store"
+        case localFile = "local-file"
```

Comment:
Do we need rawValues here?

Reply:
I think it's still the same case as in https://github.com/mapbox/mapbox-maps-ios/pull/1362#discussion_r897878636 - these are not coding keys, but rather values so they won't get automatically converted by the custom `KeyDecodingStrategy`.

Reply:
Missed the previous answer. Thanks!